EventLog.md

```diff
@@ -0,0 +1,397 @@
+
+````sql
+CREATE TYPE evt_lifecycle AS ENUM (
+    'Start',
+    'End'
+);
+
+ALTER TYPE evt_lifecycle OWNER TO postgres;
+````
+
+Which are translated into generated Scala enum types in `gen2`:
+
+````scala
+      enum("EventLifecycle") {
+        type EventLifecycleRec = Record.`'tag -> String`.T
+        val io = sql"""
+          SELECT enumlabel a, enumlabel b
+          FROM pg_enum JOIN pg_type ON pg_enum.enumtypid = pg_type.oid
+          WHERE pg_type.typname = 'evt_lifecycle'
+        """.query[(String, EventLifecycleRec)].list
+        io.transact(xa).unsafePerformIO
+      }
+````
+
+Second, the stage type itself:
+
+````sql
+CREATE TYPE evt_observe_stage AS ENUM (
+    'Sequence',
+    'Slew',
+    'Visit'
+);
+
+ALTER TYPE evt_observe_stage OWNER TO postgres;
+````
+
+with corresponding code to generate a Scala `EventObserveStage` enum.  Finally, the table itself:
+
+````sql
+CREATE TABLE log_observe_stage
+(
+   id            SERIAL,
+   "timestamp"   timestamp (5) WITH TIME ZONE NOT NULL DEFAULT CURRENT_TIMESTAMP,
+   lifecycle     evt_lifecycle NOT NULL,
+   observe_stage evt_observe_stage NOT NULL,
+   sequence_id   text NOT NULL
+);
+ 
+ALTER TABLE log_observe_stage OWNER TO postgres;
+
+ALTER TABLE ONLY log_observe_stage
+    ADD CONSTRAINT log_observe_stage_pkey PRIMARY KEY (id);
+
+CREATE INDEX ix_log_observe_stage_timestamp ON log_observe_stage USING btree ("timestamp" DESC);
+````
+
+There are a couple of things I'm not sure about here.  First, `timestamp`, even in combination with sequence id, doesn't seem like a valid primary key.  Do we need a primary key at all?  We'll never actually join on the primary key, I think. The reason I could find for it was avoiding duplicates when something goes wrong reading from a backup.
+
+Next, the timestamp itself is a `timestamp with time zone`.  It took me a while of floundering to find something that would work as expected, since I was naturally drawn to `timestamp without time zone`.  As I understand it tough, the timestamp is stored properly as an instance from a standard epoch in UTC.  The `with time zone` is just about how time input is interpreted?
+
+Finally, I made an index on the `timestamp` since that supports a major use case -- finding events between two given times.  I think we'll also want an index on the sequence id since I anticipate wanting all the events for a given sequence or observation.
+
+### 2. Observe Control Events
+
+Observe control refers to pause/continue, stop/abort actions.  The corresponding `ENUM` is:
+
+````sql
+create TYPE evt_observe_ctrl AS ENUM (
+    'Abort',
+    'Continue',
+    'Pause',
+    'Stop'
+);
+
+ALTER TYPE evt_observe_ctrl OWNER TO postgres;
+````
+
+and the table itself is
+
+````sql
+CREATE TABLE log_observe_ctrl
+(
+   id            SERIAL,
+   "timestamp"   timestamp (5) WITH TIME ZONE NOT NULL DEFAULT CURRENT_TIMESTAMP,
+   observe_ctrl  evt_observe_ctrl NOT NULL,
+   sequence_id   text NOT NULL,
+   why           text
```

Comment:
I was thinking the "why" would be typed in by a human to explain why she is pausing or aborting the observation.  That's often a weather related reason like "humidity rising" or "cloud passing" but I don't know if we can enumerate all the reasons.

Maybe it doesn't belong here though.  A nightlog app would obviously want to draw upon these same events, combined with faults and other random comments from observers.  It would need its own model though and that might be a better place for talking about "why" they are doing things.